src/element-stub.js

```diff
@@ -33,3 +43,50 @@ export class ElementStub extends BaseElement {
     return true;
   }
 }
+
+// function to calculate scriptSrc from location path
+export function calculateScript(path) {
```

Comment:
Is this exported only for testing reasons?

I'd name this: `calculateAdScriptUrl`